ci/cloudbuild/build.sh

```diff
@@ -179,10 +179,11 @@ fi
 CODECOV_TOKEN="$(tr -d '[:space:]' <<<"${CODECOV_TOKEN:-}")"
 LOG_LINKER_PAT="$(tr -d '[:space:]' <<<"${LOG_LINKER_PAT:-}")"
 
-export TRIGGER_TYPE
+export CODECOV_TOKEN
 export BRANCH_NAME
 export COMMIT_SHA
-export CODECOV_TOKEN
+export TRIGGER_TYPE
+export VERBOSE_FLAG
```

Comment:
How does even an exported VERBOSE_FLAG make it through to a cloud build?  I thought that environment was fully controlled.

Reply:
If you run `ci/cloudbuild/build.sh -t clang-tidy-pr --cloud=...` then this `ci/cloudbuild/build.sh` first runs locally, then calls `gcloud builds submit` to start "something" (I will explain in a second) in the google cloud build:

https://github.com/googleapis/google-cloud-cpp/blob/3f7dd3768dbc16c39d72bd2ec6e3c4519234018a/ci/cloudbuild/build.sh#L281

The "something" is defined here:

https://github.com/googleapis/google-cloud-cpp/blob/3f7dd3768dbc16c39d72bd2ec6e3c4519234018a/ci/cloudbuild/build.sh#L275

Which actually starts `ci/cloud/build.sh` *again*, but this time in the GCB environment:

https://github.com/googleapis/google-cloud-cpp/blob/3f7dd3768dbc16c39d72bd2ec6e3c4519234018a/ci/cloudbuild/cloudbuild.yaml#L98

That in turn calls `clang-tidy.sh`:

https://github.com/googleapis/google-cloud-cpp/blob/3f7dd3768dbc16c39d72bd2ec6e3c4519234018a/ci/cloudbuild/build.sh#L234

This time "locally" (that is local w.r.t. the GCB environment) and at that point we need the environment variables to be set.

I won't say  "hopefully that makes sense", but maybe "hopefully that explains things"?

Reply:
> Which actually starts `ci/cloud/build.sh` again, but this time in the GCB environment

Hang on.  How does this ever get a verbose flag?

Reply:
> whether it gets it or not, the `clang-tidy.sh` script still checks for the environment variable.

Then I reckon `clang-tidy.sh` would (also) do well not to depend on `VERBOSE_FLAG` even being in the environment.